sys-libs/db/db-5.3.28-r4.ebuild

```diff
@@ -0,0 +1,249 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit db flag-o-matic java-pkg-opt-2 autotools multilib-minimal toolchain-funcs
+
+#Number of official patches
+#PATCHNO=`echo ${PV}|sed -e "s,\(.*_p\)\([0-9]*\),\2,"`
+PATCHNO=${PV/*.*.*_p}
+if [[ ${PATCHNO} == "${PV}" ]] ; then
+	MY_PV=${PV}
+	MY_P=${P}
+	PATCHNO=0
+else
+	MY_PV=${PV/_p${PATCHNO}}
+	MY_P=${PN}-${MY_PV}
+fi
```

Comment:
why not using `ver_cut` ?

Reply:
To be frank I don't know quite enough of eapi7-ver/native eapi7 version functions
to change this, and I don't see much reason to. It works already, so why change
it? If there's something to be gained I'm down for that, though.

Reply:
```
PATCHNO="$(ver_cut 5)"
MY_PV="$(ver_cut 1-3)"
MY_P=${PN}-${MY_PV}

S="${WORKDIR}/${MY_P}"
....

for (( i=1 ; i<=${PATCHNO:-0} ; i++ )) ; do
	SRC_URI+=" http://www.oracle.com/technology/products/berkeley-db/db/update/${MY_PV}/patch.${MY_PV}.${i}"
done

```

seems cleaner and easier for me - why porting to eapi7 without using its features ?

Reply:
the patchname is also `quite generic`, maybe something like
` .../update/${MY_PV}/patch.${MY_PV}.${i} -> ${MY_P}_p${i}` would be more clear.

furthermore I think homepage und source uri could be changed to `https`

Reply:
> ```
> PATCHNO="$(ver_cut 5)"
> MY_PV="$(ver_cut 1-3)"
> MY_P=${PN}-${MY_PV}
> 
> S="${WORKDIR}/${MY_P}"
> ....
> 
> for (( i=1 ; i<=${PATCHNO:-0} ; i++ )) ; do
> 	SRC_URI+=" https://www.oracle.com/technology/products/berkeley-db/db/update/${MY_PV}/patch.${MY_PV}.${i}"
> done
> ```
> 
> seems cleaner and easier for me - why porting to eapi7 without using its features ?

The major feature is the cross-friendlyness, but that does appear to be enough. I'll
refactor with your suggestions as soon as I'm at my workstation.